src/BuiltInTools/dotnet-watch/HotReload/RunningProject.cs

```diff
@@ -68,5 +68,24 @@ public async ValueTask WaitForProcessRunningAsync(CancellationToken cancellation
         {
             await DeltaApplier.WaitForProcessRunningAsync(cancellationToken);
         }
+
+        public async ValueTask<int> TerminateAsync(CancellationToken shutdownCancellationToken)
+        {
+            if (shutdownCancellationToken.IsCancellationRequested)
+            {
+                // Ctrl+C sent, wait for the process to exit
+                try
+                {
+                    _ = await RunningProcess.WaitAsync(TimeSpan.FromSeconds(5), CancellationToken.None);
```

Comment:
I think we should introduce an environment variable (or however dotnet watch usually makes this) to allow users to tweak the default timeout (default would be 5 seconds).

Reply:
Will do that in a follow up.